qa/mc_sc_connected_nodes.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python2
+import json
+from SidechainTestFramework.sc_boostrap_info import SCNodeConfiguration, SCCreationInfo, MCConnectionInfo, \
+    SCNetworkConfiguration
+from SidechainTestFramework.sc_test_framework import SidechainTestFramework
+from test_framework.util import assert_equal, initialize_chain_clean, start_nodes, websocket_port, connect_nodes_bi
+from SidechainTestFramework.scutil import check_mainchan_block_inclusion, connect_sc_nodes
+
+"""
+Check the websocket connection between sidechain and mainchain nodes.
+
+Configuration: start 2 mainchain nodes and 3 sidechain nodes (with default websocket configuration) bootstrapped, 
+    respectively, from mainchain node first, first, and third. Mainchain nodes are not connected between them. 
+    Sidechain nodes are not connected between them.
+
+Test:
+    - verify genesis information for SC node 1
+    - verify genesis information for SC node 2
+    - verify genesis information for SC node 3
+    - MC 1 mine a new block
+    - verify the block is included inside SC nodes 1 and 2
+    - verify the block is NOT included inside SC node 3
+    - connect MC 1 to MC 2
+    - connect SC 1 to SC 2 and SC 3
+    - verify the block is included inside SC node 3
+    - MC 2 mine a new block
+    - verify the block is included inside SC nodes 1, 2 and 3
+"""
+class MCSCConnectedNodes(SidechainTestFramework):
+
+    number_of_mc_nodes = 2
+    number_of_sidechains = 3
+    ws_port_mc_0 = websocket_port(0)
+    ws_port_mc_1 = websocket_port(1)
+    sc_nodes_bootstrap_info=None
+
+    def setup_chain(self):
+        initialize_chain_clean(self.options.tmpdir, self.number_of_mc_nodes, [self.ws_port_mc_0, self.ws_port_mc_1])
+
+    def setup_network(self, split = False):
+        self.nodes = self.setup_nodes()
+        self.sync_all()
+
+    def setup_nodes(self):
+        return start_nodes(self.number_of_mc_nodes, self.options.tmpdir, extra_args=[["-websocket"],["-websocket"],["-websocket"]])
+
+    def sc_setup_chain(self):
+        mc_node_1 = self.nodes[0]
+        mc_node_2 = self.nodes[1]
+        sc_node_1_configuration = SCNodeConfiguration(
+            MCConnectionInfo(address="ws://{0}:{1}".format(mc_node_1.hostname, self.ws_port_mc_0))
+        )
+        sc_node_2_configuration = SCNodeConfiguration(
+            MCConnectionInfo(address="ws://{0}:{1}".format(mc_node_1.hostname, self.ws_port_mc_0))
+        )
+        sc_node_3_configuration = SCNodeConfiguration(
+            MCConnectionInfo(address="ws://{0}:{1}".format(mc_node_2.hostname, self.ws_port_mc_1))
+        )
+        network = SCNetworkConfiguration(SCCreationInfo(mc_node_1, "1".zfill(64), 600, 1000),
+                                         sc_node_1_configuration, sc_node_2_configuration, sc_node_3_configuration)
+        self.sc_nodes_bootstrap_info = self.bootstrap_sidechain_nodes(network)
+
+    def sc_setup_network(self, split=False):
```

Comment:
instead of overriding the `sc_setup_network` in this case, `sc_setup_nodes` should be changed. Moreover `sc_setup_nodes` take no args similar to `setup_nodes`.

Reply:
Done